lib/monitor.js

```diff
@@ -192,8 +192,10 @@ class Monitor extends Events.EventEmitter {
             this._server.removeListener(event, handler);
         });
 
-        this.emit('stop');
-        return callback();
+        this._dataStream.push(null);
+        // Do a nextTick here so that all the streams listening for "end" have a chance to run
+        // https://github.com/nodejs/node/blob/master/lib/_stream_readable.js#L894-L897
+        return process.nextTick(callback);
```

Comment:
I was just flailing tying to figure out why the "end" event wasn't happening, when I added this the "end" event consistently happens. Since technically IO could be happening in the streams, setImmediate makes sense.